python_modules/libraries/dagster-k8s/dagster_k8s/executor.py

```diff
@@ -220,26 +220,20 @@ def launch_step(self, step_handler_context: StepHandlerContext):
             },
         )
 
-        events.append(
-            DagsterEvent(
-                event_type_value=DagsterEventType.ENGINE_EVENT.value,
-                pipeline_name=step_handler_context.pipeline_run.pipeline_name,
-                step_key=step_key,
-                message=f"Executing step {step_key} in Kubernetes job {job_name}",
-                event_specific_data=EngineEventData(
-                    [
-                        MetadataEntry("Step key", value=step_key),
-                        MetadataEntry("Kubernetes Job name", value=job_name),
-                    ],
-                ),
-            )
+        yield DagsterEvent.engine_event(
```

Comment:
definitely an improvement to log these events immediately, instead of once the fn returned like before